components/OAuth2.0/authorization.c

```diff
@@ -0,0 +1,282 @@
+                SaveFileInSpiffsWithTxtFormat(AuthInterfaceHandler->ConfigAddressInSpiffs,                              // Save new file in the directory
+                                              "refresh_token", AuthInterfaceHandler->token.RefreshToken, 
+                                              NULL, NULL);
+                xSemaphoreGive((*AuthInterfaceHandler->IsServiceAuthorizedSemaphore));                                  // give IsServiceAuthorizedSemaphore semaphore in the IntefaceHandler                
+                AuthInterfaceHandler->Status = CHECK_TIME;                                                                 // set Status to CHECK_TIME     
+                break;
+            }
+            case CHECK_TIME:
+            {
+                bool IsTokenExpired = Oauth_IsTokenExpired();                                                               // Check if the expiration time has elapsed since the last received token
+                if (!IsTokenExpired)
+                {
+                    // keep state
+                    break;
+                }  
+                AuthInterfaceHandler->Status = EXPIRED;                                                                // set Status to EXPIRED if token expired
+          
+                break;
+            }
+            case EXPIRED:
+            {
+                ESP_LOGW(TAG, "token is expired");
+                bool IsTokenRenewed = Oauth_TokenRenew();
+                if (!IsTokenRenewed)                                                                   // Run function to renew the token
+                {
+                    AuthInterfaceHandler->Status = LOGIN;
+                    break;                                                                   // set Status to ILDLE if token renewed unsuccessfully
+                }
+
+                xSemaphoreGive((*AuthInterfaceHandler->IsServiceAuthorizedSemaphore));                              // give IsServiceAuthorizedSemaphore semaphore in the IntefaceHandler
+                AuthInterfaceHandler->TokenLastUpdate = xTaskGetTickCount();                                           // Save the time when the token was received
+                AuthInterfaceHandler->Status = CHECK_TIME;                                                             // set Status to CHECK_TIME if token renewed successfully
+                break;
+            }
+        }   
+        vTaskDelay(pdMS_TO_TICKS(100)); //FIXME magic number                                                                            // Task delay at the end of while(1) loop
+    }
+}
+
+/**
+ * @brief This function initiates the authorization process.
+ * @param InterfaceHandler as the handler
+ * @return true if task run to the end
+ */
+bool Oauth_TaskInit(OAuthInterfaceHandler_t *InterfaceHandler)
+{
+    AuthInterfaceHandler = InterfaceHandler;
+    AuthInterfaceHandler->Status = INIT;
```

Comment:
the handler that holds private variables such as `Status` should be same as the handler that is exposed in the API header file.

Reply:
fixed.